WalletWasabi/WebClients/BlockstreamInfo/BlockstreamInfoClient.cs

```diff
@@ -0,0 +1,88 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Blockchain.Mempool;
+using WalletWasabi.Tor.Http.Extensions;
+
+namespace WalletWasabi.WebClients.BlockstreamInfo
+{
+	public class BlockstreamInfoClient : IMempoolSyncer
+	{
+		private bool _disposedValue;
+
+		public BlockstreamInfoClient(Network network)
+		{
+			var baseUrl = "https://blockstream.info/";
+			if (network == Network.Main)
+			{
+				baseUrl += "api/";
+			}
+			else if (network == Network.TestNet)
+			{
+				baseUrl += "testnet/api/";
+			}
+			else
+			{
+				throw new NotSupportedException("Specified network not supported.");
+			}
+
+			HttpClient = new HttpClient
```

Comment:
https://github.com/zkSNACKs/WalletWasabi/pull/4792#issue-526761456 should be taken into account